lib/compat/wordpress-6.6/blocks.php

```diff
@@ -44,3 +44,221 @@ function gutenberg_replace_pattern_override_default_binding( $parsed_block ) {
 }
 
 add_filter( 'render_block_data', 'gutenberg_replace_pattern_override_default_binding', 10, 1 );
+
+// Only process caption in WordPress 6.5.
+if ( ! is_wp_version_compatible( '6.6' ) && is_wp_version_compatible( '6.5' ) ) {
+	/**
+	 * Replace the caption value in the HTML based on the binding value.
+	 *
+	 * @param string $block_content  Block Content.
+	 * @param string $block_name     The name of the block to process.
+	 * @param string $attribute_name The attribute name to replace.
+	 * @param mixed  $source_value   The value used to replace in the HTML.
+	 * @return string The modified block content.
+	 */
+	function gutenberg_block_bindings_replace_caption( $block_content, $block_name, string $attribute_name, $source_value ) {
+		// Create private anonymous class until the HTML API provides `set_inner_html` method.
+		$block_reader = new class($block_content) extends WP_HTML_Tag_Processor{
+			/**
+			 * THESE METHODS ARE A TEMPORARY SOLUTION NOT TO BE EMULATED.
+			 * IT IS A TEMPORARY SOLUTION THAT JUST WORKS FOR THIS SPECIFIC
+			 * USE CASE UNTIL THE HTML PROCESSOR PROVIDES ITS OWN METHOD.
+			 */
+
+			/**
+			 * Replace the inner text of an HTML with the passed content.
+			 *
+			 * @param string $new_content New text to insert in the HTML element.
+			 * @return bool Whether the inner text was properly replaced.
+			 */
+			public function gutenberg_set_inner_text( $new_content ) {
+				/*
+				 * THIS IS A STOP-GAP MEASURE NOT TO BE EMULATED.
+				 *
+				 * Check that the processor is paused on an opener tag.
+				 *
+				 */
+				if (
+					WP_HTML_Tag_Processor::STATE_MATCHED_TAG !== $this->parser_state ||
+					$this->is_tag_closer()
+				) {
+					return false;
+				}
+
+				// Set position of the opener tag.
+				$this->set_bookmark( 'opener_tag' );
+
+				/*
+				 * This is a best-effort guess to visit the closer tag and check it exists.
+				 * In the future, this code should rely on the HTML Processor for this kind of operation.
+				 */
+				$tag_name = $this->get_tag();
+				if ( ! $this->next_tag(
+					array(
+						'tag_name'    => $tag_name,
+						'tag_closers' => 'visit',
+					)
+				) || ! $this->is_tag_closer() ) {
+					return false;
+				}
+
+				// Set position of the closer tag.
+				$this->set_bookmark( 'closer_tag' );
+
+				// Get opener and closer tag bookmarks.
+				$opener_tag_bookmark = $this->bookmarks['opener_tag'];
+				$closer_tag_bookmark = $this->bookmarks['closer_tag'];
+
+				// Appends the new content.
+				$after_opener_tag = $opener_tag_bookmark->start + $opener_tag_bookmark->length;
+				/*
+				 * There was a bug in the HTML Processor token length fixed after 6.5.
+				 * This check is needed to add compatibility for that.
+				 * Related issue: https://github.com/WordPress/wordpress-develop/pull/6625
+				 */
+				if ( '>' === $this->html[ $after_opener_tag ] ) {
+					++$after_opener_tag;
+				}
```

Comment:
We need to be very careful that this check is correct.

The tag opener _must_ end in `>`:

```html
<div att="foo">bla
```

But the content inside the tag could start with `>`!

```html
<div att="foo">> this starts with a &gt; character!
```

Reply:
You're right! I just changed it in the latest commit.

I think it was working on my testing because it seems to be transformed to `&xxx`. But checking the ending seems safer.